checks/phpcs.php

```diff
@@ -110,21 +110,133 @@ protected function phpcs_result_to_warnings( $result ) {
 				}
 
 				$source_code = esc_html( trim( file( $this->path . '/' . $filename )[ $message['line'] - 1 ] ) );
+				$edit_link   = sprintf(
+					'<a href="%1$s" title="%2$s" aria-label="%2$s" target="_blank">%3$s</a>',
+					$this->get_file_editor_url( $filename, $message['line'] ),
+					sprintf(
+						/* translators: %s is the path to a plugin file. */
+						esc_attr__( 'View %s in the plugin file editor.', 'plugin-check' ),
+						$this->slug . '/' . $filename
+					),
+					esc_html__( 'View in code editor', 'plugin-check' )
+				);
```

Comment:
This needs a capability check as not every WordPress setup allows editing plugin files and not every user may have the permissions to it.
```suggestion
				if ( current_user_can( 'edit_plugins' ) {
					$edit_link   = sprintf(
						'<a href="%1$s" title="%2$s" aria-label="%2$s" target="_blank">%3$s</a>',
						$this->get_file_editor_url( $filename, $message['line'] ),
						sprintf(
							/* translators: %s is the path to a plugin file. */
							esc_attr__( 'View %s in the plugin file editor.', 'plugin-check' ),
							$this->slug . '/' . $filename
						),
						esc_html__( 'View in code editor', 'plugin-check' )
					);
				}
```